### PR TITLE
Update sprintf and vsprintf to make use of the sprintf-js package.

### PR DESCRIPTION
**Pull request by jtangelder**
The owner (@alexei) of the current implementation of sprintf has continued development and improved the implementation. It makes sense to use of his package on npm instead of using an out-of-date copy.

### CODE REVIEW

**Comment by stoeffel** [comment]
I would vote for removing `sprintf` completely and just add a link to @alexei's module. Less code to maintain and we don't have to copy the changes from the origin to our module. @epeli do you agree?

**Comment by epeli** [comment]
Yes. We should add a warning and a link when this function is being used in next minor release and then completely remove it in next major.

**Comment by stoeffel** [comment]
@jtangelder could you add a warning to your pr?

**Comment by jtangelder** [comment]
Sure, do you have a function for this?

**Comment by stoeffel** [comment]
I would use https://www.npmjs.com/package/util-deprecate. @epeli also good with this?

**Comment by epeli** [comment]
yep

**Comment by jtangelder** [comment]
OK, will add it tomorrow!

**Comment by stoeffel** [comment]
Cool, thank you!


Jorik Tangelder <notifications@github.com> schrieb am Mo., 18. Jan. 2016 um
21:06 Uhr:

**Comment by jtangelder** [comment]
I've updated the PR

**Comment by stoeffel** [comment]
@jtangelder thanks. We should probably add a warning (and link) to the README as well. Sorry that I didn't mention that in my comment before :bow:

**Comment by jtangelder** [comment]
No problem, added it to the previous commit :dancer:

File: README.markdown [hunk 1]
```diff
@@ -530,11 +530,9 @@ words("   ")
 
 #### sprintf(string format, ...arguments) => string
 
-C like string formatting.
-Credits goes to [Alexandru Marasteanu][o].
-For more detailed documentation, see the [original page][o].
+C like string formatting. Makes use of the [sprintf-js](https://npmjs.org/package/sprintf-js) package.
 
-[o]: http://www.diveintojavascript.com/projects/javascript-sprintf
+**This function will be removed the next major release, use the [sprintf-js](https://npmjs.org/package/sprintf-js) package instead.**
```
Comment (stoeffel) [inline README.markdown]:
```diff
-This function will be removed the next major release
+This function will be removed in the next major release
```

File: sprintf.js [hunk 2]
```diff
@@ -1,124 +1,4 @@
...
-    }
-    return parse_tree;
-  };
-
-  return str_format;
-})();
-
-module.exports = sprintf;
+module.exports = deprecate(require('sprintf-js').sprintf,
+  'sprintf() is will be removed the next major release, use the sprintf-js package instead.');
```
Comment (stoeffel) [inline sprintf.js]:
```diff
-sprintf() is will be
+sprintf() will be
```

File: vsprintf.js [hunk 3]
```diff
@@ -1,6 +1,4 @@
-var sprintf = require('./sprintf');
+var deprecate = require('util-deprecate');
 
-module.exports = function vsprintf(fmt, argv) {
-  argv.unshift(fmt);
-  return sprintf.apply(null, argv);
-};
+module.exports = deprecate(require('sprintf-js').vsprintf,
+  'vsprintf() is will be removed the next major release, use the sprintf-js package instead.');
```
Comment (stoeffel) [inline vsprintf.js]:
```diff
-sprintf() is will be
+sprintf() will be
```

**Comment by stoeffel** [comment]
> No problem, added it to the previous commit :dancer:

Nice.

There are two minor typos left. Could you fix them an squash your commits.
Then it's good to :ship:. Thanks

**Comment by jtangelder** [comment]
Fixed them!

File: sprintf.js [hunk 4]
```diff
@@ -1,124 +1,4 @@
...
-    }
-    return parse_tree;
-  };
-
-  return str_format;
-})();
-
-module.exports = sprintf;
+module.exports = deprecate(require('sprintf-js').sprintf,
+  'sprintf() will be removed the next major release, use the sprintf-js package instead.');
```
Comment (stoeffel) [inline sprintf.js]: still missing `in`
`will be removed in the`

File: vsprintf.js [hunk 5]
```diff
@@ -1,6 +1,4 @@
-var sprintf = require('./sprintf');
+var deprecate = require('util-deprecate');
 
-module.exports = function vsprintf(fmt, argv) {
-  argv.unshift(fmt);
-  return sprintf.apply(null, argv);
-};
+module.exports = deprecate(require('sprintf-js').vsprintf,
+  'vsprintf() will be removed the next major release, use the sprintf-js package instead.');
```
Comment (stoeffel) [inline vsprintf.js]: still missing `in`
`will be removed in the`

**Comment by stoeffel** [comment]
arg didn't see the missing `in` until now. Sorry, could you fix that as well?

**Comment by jtangelder** [comment]
Got it ^

**Comment by stoeffel** [comment]
Awesome thanks.

**Pull request merged**